crypto/conf/conf_def.c

```diff
@@ -555,13 +614,94 @@ static int str_copy(CONF *conf, char *section, char **pto, char *from)
     return 0;
 }
 
+static BIO *process_include(char *include, OPENSSL_DIR_CTX **dirctx,
+                            char **dirpath)
+{
+    struct stat st = { 0 };
+    BIO *next;
+
+    if (stat(include, &st) < 0) {
+        SYSerr(SYS_F_STAT, errno);
+        ERR_add_error_data(1, include);
+        /* missing include file is not fatal error */
+        return NULL;
+    }
+
+    if ((st.st_mode & S_IFDIR) == S_IFDIR) {
+        if (*dirctx != NULL) {
+            CONFerr(CONF_F_PROCESS_INCLUDE,
+                    CONF_R_RECURSIVE_DIRECTORY_INCLUDE);
+            ERR_add_error_data(1, include);
+            return NULL;
+        }
+        if ((next = get_next_file(include, dirctx)) != NULL)
+            *dirpath = include;
+        return next;
+    }
+
+    next = BIO_new_file(include, "r");
+    return next;
+}
+
+static BIO *get_next_file(const char *path, OPENSSL_DIR_CTX **dirctx)
+{
+    const char *filename;
+
+    while ((filename = OPENSSL_DIR_read(dirctx, path)) != NULL) {
+        size_t namelen;
+
+        namelen = strlen(filename);
+
+        if ((namelen > 5 && strcasecmp(filename + namelen - 5, ".conf") == 0)
+            || (namelen > 4 && strcasecmp(filename + namelen - 4, ".cnf") == 0)) {
+            size_t newlen;
+            char *newpath;
+            BIO *bio;
+
+            newlen = strlen(path) + namelen + 2;
+            newpath = OPENSSL_zalloc(newlen);
+            if (newpath == NULL) {
+                CONFerr(CONF_F_GET_NEXT_FILE, ERR_R_MALLOC_FAILURE);
+                break;
+            }
+            OPENSSL_strlcat(newpath, path, newlen);
+#ifndef OPENSSL_SYS_VMS
+            OPENSSL_strlcat(newpath, "/", newlen);
+#endif
```

Comment:
But it's hardly appropriate to simply append names on VMS. I mean I'm pretty sure it wouldn't work. [But ping @levitte.] Without #else if would probably be more honest if feature was not available on VMS...

Reply:
I copied this behavior from ssl/ssl_cert.c - unfortunately I have no idea how VMS should be handled. @levitte either please suggest how to handle it or it can be fixed with further commit later?

Reply:
As long as `path` is the full directory spec needed (`vol:[dir.dir.dir]` when absolute or `[.dir]` when relative), just adding `filename` i exactly the right thing to do.  As far as I can see, this is the case here...

Reply:
Ah! In other words `path` is always `[some.thing]`, i.e. in square brackets. OK. Sorry about confusion.